app/services/schools/register_mentor.rb

```diff
@@ -41,5 +45,17 @@ def school
     def start_at_school!
       teacher.mentor_at_school_periods.create!(school:, started_on:, email:)
     end
+
+    def early_roll_out_mentor?
+      EarlyRollOutMentor.find_by(trn:).present?
+    end
+
+    def mentor_ineligible_for_funding_reason
+      'completed_during_early_roll_out' if early_roll_out_mentor?
+    end
+
+    def mentor_funding_end_date
+      Date.new(2021, 4, 19) if early_roll_out_mentor?
+    end
```

Comment:
IMO we should move this logic to the wizard Mentor class and pass the resulting values (`mentor_ineligible_for_funding_reason` and `mentor_funding_end_date`) in the call to this service object. The reasons for it are:

- the Mentor class is also searching the `EarlyRollOutMentor` table to determine if the mentor in registration is training fundable or not, so better not to have logic duplicated.

- this service object is only to be consumed by the `RegisterMentor` journey because the context is so unique (loads of validations and checks have been done in the steps) that it can't be the case anywhere else in the service.

- these are also properties of the mentor in registration and we might need that date or reason in the journey.